doc/user-guide/dask.rst

```diff
@@ -2,195 +2,151 @@
 
-    os.remove("manipulated-example-data.nc")  # Was not opened.
+.. tab:: GeoTIFF
 
-.. note::
+    Open large geoTIFF files with rioxarray::
 
-    When using Dask's distributed scheduler to write NETCDF4 files,
-    it may be necessary to set the environment variable `HDF5_USE_FILE_LOCKING=FALSE`
-    to avoid competing locks within the HDF5 SWMR file locking scheme. Note that
-    writing netCDF files with Dask's distributed scheduler is only supported for
-    the `netcdf4` backend.
+        xds = rioxarray.open_rasterio("my-satellite-image.tif", chunks='auto')
 
-A dataset can also be converted to a Dask DataFrame using :py:meth:`~xarray.Dataset.to_dask_dataframe`.
+    See :ref:`io.rasterio` for more details.
 
-.. ipython:: python
-    :okwarning:
 
-    df = ds.to_dask_dataframe()
-    df
+Loading Dask Arrays
+~~~~~~~~~~~~~~~~~~~
 
-Dask DataFrames do not support multi-indexes so the coordinate variables from the dataset are included as columns in the Dask DataFrame.
+.. ipython:: python
+    :suppress:
 
+    import os
 
-Using Dask with xarray
-----------------------
+    import numpy as np
+    import pandas as pd
+    import xarray as xr
+
+    np.random.seed(123456)
+    np.set_printoptions(precision=3, linewidth=100, threshold=100, edgeitems=3)
 
-Nearly all existing xarray methods (including those for indexing, computation,
-concatenating and grouped operations) have been extended to work automatically
-with Dask arrays. When you load data as a Dask array in an xarray data
-structure, almost all xarray operations will keep it as a Dask array; when this
-is not possible, they will raise an exception rather than unexpectedly loading
-data into memory. Converting a Dask array into memory generally requires an
-explicit conversion step. One notable exception is indexing operations: to
-enable label based indexing, xarray will automatically load coordinate labels
-into memory.
+    ds = xr.Dataset(
+        {
+            "temperature": (
+                ("time", "latitude", "longitude"),
+                np.random.randn(30, 180, 180),
+            ),
+            "time": pd.date_range("2015-01-01", periods=30),
+            "longitude": np.arange(180),
+            "latitude": np.arange(89.5, -90.5, -1),
+        }
+    )
+    ds.to_netcdf("example-data.nc")
 
-.. tip::
+There are a couple common cases where you may want to convert lazy Dask arrays into eager, in-memory Xarray data structures:
 
-   By default, dask uses its multi-threaded scheduler, which distributes work across
-   multiple cores and allows for processing some datasets that do not fit into memory.
-   For running across a cluster, `setup the distributed scheduler <https://docs.dask.org/en/latest/setup.html>`_.
+- You want to inspect smaller intermediate results when working interactively or debugging
+- You've reduced the dataset (by filtering or with a groupby, for example) and now have something much smaller that fits in memory
```

Comment:
To this, I would add:

- Computing intermediate results is a useful escape hatch for cases where Dask is unable (or struggles) to process something in a streaming fashion. The canonical example of this is normalizing a dataset, e.g., `ds - ds.mean()`, when `ds` is larger than memory. Typically, you should either save `ds` to disk or compute `ds.mean()` eagerly.

Reply:
Added!